runtime/compiler/env/j9methodServer.cpp

```diff
@@ -351,54 +368,32 @@ TR_ResolvedJ9JITServerMethod::getResolvedPossiblyPrivateVirtualMethod(TR::Compil
          !comp->ilGenRequest().details().isMethodHandleThunk() && // cmvc 195373
          performTransformation(comp, "Setting as unresolved virtual call cpIndex=%d\n",cpIndex) ) || ignoreRtResolve || shouldCompileTimeResolve)
       {
-      _stream->write(JITServer::MessageType::ResolvedMethod_getResolvedPossiblyPrivateVirtualMethodAndMirror, (TR_ResolvedMethod *) _remoteMirror, literals(), cpIndex);
-      auto recv = _stream->read<J9Method *, UDATA, bool, TR_ResolvedJ9JITServerMethodInfo>();
-      J9Method *ramMethod = std::get<0>(recv);
-      UDATA vTableIndex = std::get<1>(recv);
-      bool isUnresolvedInCP = std::get<2>(recv);
+      _stream->write(JITServer::MessageType::ResolvedMethod_getResolvedPossiblyPrivateVirtualMethodAndMirror, _remoteMirror, cpIndex, ignoreRtResolve);
+      auto recv = _stream->read<TR_OpaqueMethodBlock *, TR_ResolvedJ9JITServerMethodInfo, uint32_t, bool>();
+      TR_OpaqueMethodBlock *ramMethod = std::get<0>(recv);
+      auto methodInfo = std::get<1>(recv);
+      uint32_t vTableSlot = std::get<2>(recv);
+      bool isUnresolvedInCP = std::get<3>(recv);
 
       if (unresolvedInCP)
          *unresolvedInCP = isUnresolvedInCP;
 
-      bool createResolvedMethod = true;
-
-      if (comp->compileRelocatableCode() && ramMethod && comp->getOption(TR_UseSymbolValidationManager))
+      if (verifyCaching)
          {
-         if (!comp->getSymbolValidationManager()->addVirtualMethodFromCPRecord((TR_OpaqueMethodBlock *)ramMethod, cp(), cpIndex))
-            createResolvedMethod = false;
+         TR_ASSERT_FATAL(verifyCache(methodInfo, cachedMethodInfo, unresolvedInCache, unresolvedInCP), "Cache and client method info did not match");
```

Comment:
Note that this check will be run even if server didn't have anything cached. This should be functionally correct because `cachedMethodInfo` will be left in the default state, and `cachedMethodInfo.remoteMirror == NULL` by default, so the verification will trivially succeed, but it's something to keep in mind.